backend/fleet_management/crypto.py

```diff
@@ -0,0 +1,104 @@
+from math import floor, sqrt
+from secrets import randbits
+from collections import namedtuple
+
+from django.conf import settings
+
+
+PublicKey = namedtuple("PublicKey", "n e")
+PrivateKey = namedtuple("PrivateKey", "n d")
+
+
+def sign(message: int, priv: PrivateKey) -> int:
+    return pow(message, priv.d, priv.n)
+
+
+def verify(message: int, signature: int, pub: PublicKey) -> bool:
+    return pow(signature, pub.e, pub.n) == message % pub.n
+
+
+def inverse_of(a: int, b: int) -> int:
+    """Returns n^-1 (mod p)."""
+    x0, x1, y0, y1 = 1, 0, 0, 1
+    oa, ob = a, b
+
+    while b != 0:
+        q = a // b
+        a, b = b, a % b
+        x0, x1 = x1, x0 - q * x1
+        y0, y1 = y1, y0 - q * y1
+
+    if x0 < 0:
+        x0 += ob
+
+    if y0 < 0:
+        y0 += oa
+
+    # when a and b are not relatively prime
+    if a != 1 or (oa * x0) % ob != 1:
```

Comment:
That's rather `c` (or `golang`) style. Python should raise exception here.

Reply:
(or is that proper result based on module algebra?)

Reply:
I tried to avoid throwing any exceptions in math functions to keep stuff bit simpler.
Technically prime numbers can be negative which would imply that the -1 could be one of possible outcomes of this function, so maybe the exception would be more sensible, but this function is used only here.